client.py

```diff
@@ -1,23 +1,76 @@
 #!/usr/bin/python
+import base64
+import math
+import requests
 from time import gmtime, strftime, sleep
+try:
+    from local_settings import *
+except ImportError:
+    pass
 
-# Return base64 encoded file
-def file_to_base64():
-    return ""
+class Overlord:
 
-# Send replay to an Overmind
-def send_replay():
-    pass
+    # Main constructor
+    def __init__(self, replays_url):
+        self.replays_url = replays_url
+        self.log_file = "Overlord.log"
+
+    # Send replay to an Overmind
+    def send_replay(self, replay_title, base64_encoded_text, extra_text):
+
+        # Build JSON body
+        body = {'title': replay_title, 'base64_file': base64_encoded_text, 'extra': extra_text}
+
+        # Send replay (HTTP post with exp. regression)
+        delay = 1
+        while (delay > 0):
+            try:
+                request = requests.post(self.replays_url, json=body)
+                delay = 0
+            except Exception as e:
+                sleep(math.exp(delay))
```

Comment:
Generally we set a maximum number for delay, from that number on the delay doesn't increase, and the base of the regression generally is 2